### PR TITLE
Update Installation Docs for Consul Terraform Sync

### PR DESCRIPTION
**Pull request by lornasong**
- Remove Install Consul in install docs. Will move to requirements docs
 - Add Connect your Consul Cluster in install docs
 - Add Connect your Network Device & Multi-Provider in install docs
 - Add additional multi provider details in config docs
 - Fix a case typo for Consul catalog in tasks doc

### CODE REVIEW

**Comment by lornasong** [comment]
To view "Consul catalog" change: https://deploy-preview-8881--consul-docs-preview.netlify.app/docs/nia/tasks#tasks

Screen Shot 2020-10-08 at 12 11 09 PM

**Comment by lornasong** [comment]
To view additional info on Multi providers configuration: https://deploy-preview-8881--consul-docs-preview.netlify.app/docs/nia/installation/configuration#multiple-provider-configurations

**Comment by lornasong** [comment]
To view install docs: https://deploy-preview-8881--consul-docs-preview.netlify.app/docs/nia/installation/install

See "Installing Consul" & "Connect your Network Device" sections

File: website/pages/docs/nia/installation/install.mdx [hunk 1]
```diff
@@ -6,13 +6,20 @@ description: >-
...
+## Installing Consul
 
-# Installing Consul-Terraform-Sync
+Consul-Terraform-Sync is a daemon that runs alongside Consul, similar to other Consul ecosystem tools like Consul Template. Consul-Terraform-Sync is not included with the Consul binary and needs to be installed separately.
 
-To install Consul-Terraform-Sync, find the [appropriate package](https://releases.hashicorp.com/) for your system and download it as a zip archive. Unzip the package to extract the binary named consul-terraform-sync. Move the consul-terraform-sync binary to a location available on your PATH.
+To install a local Consul agent, refer to the [Getting Started: Install Consul Tutorial](https://learn.hashicorp.com/tutorials/consul/get-started-install).
+
+The Consul agent must be running in order to dynamically update network devices. To run the local Consul agent, for simplicity, run in development mode which can be started with `consul agent -dev`. For more details on running Consul, see [Run a Consul Cluster](/docs/nia/installation/requirements#run-a-consul-cluster).
```
Comment (findkim) [inline website/pages/docs/nia/installation/install.mdx:17]: This reads a bit too strong that we expect running Consul in dev mode, when it can be an option or connect to an already existing Consul cluster. Can you revert closer to the original verbiage?

File: website/pages/docs/nia/installation/install.mdx [hunk 2]
```diff
@@ -40,6 +47,22 @@ Usage of consul-terraform-sync:
       Print the version of this daemon.
 ```
 
-# Connect your Network Device
+## Connect your Network Device
+
+Consul-Terraform-Sync interacts with your network device through a network driver. For the Terraform network driver, Consul-Terraform-Sync uses Terraform providers to make changes to your network infrastructure resources. You will need to [search for an existing](/docs/nia/installation/requirements#finding-and-using-terraform-providers) Terraform Provider for each of your network devices or [create a new one](/docs/nia/installation/requirements#how-to-create-a-provider).
```
Comment (findkim) [inline website/pages/docs/nia/installation/install.mdx:52]: Suggestion to reword the demand into a suggestion: 
> You will need to search for an existing ...

You can reference provider docs on the Terraform Registry to configure each provider or create a new Terraform provider

File: website/pages/docs/nia/installation/install.mdx [hunk 3]
```diff
@@ -40,6 +47,22 @@ Usage of consul-terraform-sync:
...
+  password = "password123"
+}
+```
+
+This provider is then used by task(s) to execute a Terraform module that will update the related network device.
+
+### Multiple Instances per Provider
 
-## Multiple Instances per Provider
+You might have multiple instances of the same type of network device; for example, multiple instances of a firewall or load balancer. You can configure each instance with its own provider block and distinguish it by the `alias` field. See [multiple provider configurations](/docs/nia/installation/configuration#multiple-provider-configurations) for more details and an example of the configuration.
```
Comment (findkim) [inline website/pages/docs/nia/installation/install.mdx:68]: Let's try to adhere closely to Terraform verbiage: field -> meta-argument ([source](https://www.terraform.io/docs/configuration/providers.html#alias-multiple-provider-configurations))

File: website/pages/docs/nia/installation/configuration.mdx [hunk 4]
```diff
@@ -227,7 +227,7 @@ task {
...
+Consul-Terraform-Sync supports the [Terraform feature to define multiple configurations](https://www.terraform.io/docs/configuration/providers.html#alias-multiple-provider-configurations) for the same provider by utilizing the `alias` meta-argument. Define multiple provider blocks with the same provider name and set the `alias` to a unique value across a given provider. Select which provider configuration to use for a task by specifying the configuration with the provider name and alias (`<name>.<alias>`) within the list of providers in the [`task.provider`](#task) parameter. Although multiple different providers are allowed per task, only one provider instance of a type of provider is allowed per task.
```
Comment (findkim) [inline website/pages/docs/nia/installation/configuration.mdx:230]: This was difficult to digest for me, a lot of qualifiers to keep straight: "multiple different" "per" "only one ... type", "per". I do like the goal of extra clarity here, however could you rephrase it?

Took a jab, feel free to take or another angle!

> A task can use multiple providers, but only one provider instance of a provider is allowed per task

**Review by findkim — COMMENTED** [review]
These additions are looking great. And makes sense as to how you're referencing other parts of the doc. I left a few suggestions to clarify / or soften the wording.

File: website/pages/docs/nia/installation/install.mdx (hunk 1 above)
Comment (findkim) [reply website/pages/docs/nia/installation/install.mdx:17]: Thinking about this at a high level, I think installing Consul is a weird first step for the Install page for Sync. Consul agent is a requirement. Can you move this portion to fit the req page instead?

Comment (lornasong) [reply website/pages/docs/nia/installation/install.mdx:17]: I see what you're saying. This is the installing sync page vs. installing everything page. I'll move it. Just to continue the conversation about the content though, what are your thoughts on:

> To run the local Consul agent, you can run Consul in development mode which can be started with `consul agent -dev` for simplicity.

Re: "Connect to your Consul cluster". I think that makes sense. I'll add it.

File: website/pages/docs/nia/installation/configuration.mdx (hunk 4 above)
Comment (lornasong) [reply website/pages/docs/nia/installation/configuration.mdx:230]: Agree, I didn't love how it turned out after a couple read. I like your suggestion, thank you!

File: website/pages/docs/nia/installation/install.mdx (hunk 1 above)
Comment (findkim) [reply website/pages/docs/nia/installation/install.mdx:17]: Yeah that reads well to me!

Sweet, thanks for looking into shifting things around -- I know you were following the original organization.

**Review by findkim — APPROVED** [review]
✨

**Review by eikenb — APPROVED** [review]
Just kind of skimmed it (as Kim already reviewed it more thoroughly) and things looked good. :+1:

**Comment by lornasong** [comment]
@findkim thanks for all of your feedback! @eikenb thanks for reviewing!
